services/src/main/java/org/keycloak/common/util/Resteasy.java

```diff
@@ -0,0 +1,156 @@
+                    ResteasyContext.clearContextData();
+                }
+            };
+        } else {
+            PUSH_CONTEXT = new BiConsumer<Class, Object>() {
+                @Override
+                public void accep(Class p1, Object p2) {
+                    ResteasyProviderFactory.getInstance().pushContext(p1, p2);
+                }
+            };
+            PUSH_DEFAULT_OBJECT = new BiConsumer<Class, Object>() {
+                @Override
+                public void accep(Class p1, Object p2) {
+                    ResteasyProviderFactory.getInstance().getContextData(Dispatcher.class).getDefaultContextObjects()
+                            .put(p1, p2);
+                }
+            };
+            PULL_CONTEXT = new Function<Class, Object>() {
+                @Override
+                public Object apply(Class p1) {
+                    return ResteasyProviderFactory.getInstance().getContextData(p1);
+                }
+            };
+            CLEAR_CONTEXT = new Runnable() {
+                @Override
+                public void run() {
+                    ResteasyProviderFactory.getInstance().clearContextData();
+                }
+            };
+        }
+    }
+
+    /**
+     * Push the given {@code instance} with type/key {@code type} to the Resteasy context associated with the current thread.
+     * 
+     * @param type the type/key to associate the {@code instance} with 
+     * @param instance the instance
+     */
+    public static void pushContext(Class type, Object instance) {
+        PUSH_CONTEXT.accep(type, instance);
+    }
+
+    /**
+     * Lookup the instance associated with the given type/key {@code type} from the Resteasy context associated with the current thread.
+     *
+     * @param type the type/key to lookup 
+     * @return the instance associated with the given {@code type} or null if non-existent.                
+     */
+    public static <R> R getContextData(Class<R> type) {
+        return (R) PULL_CONTEXT.apply(type);
+    }
+
+    /**
+     * Clear the Resteasy context associated with the current thread.
+     */
+    public static void clearContextData() {
+        CLEAR_CONTEXT.run();
+    }
+
+    /**
+     * Push the given {@code instance} with type/key {@code type} to the Resteasy global context.
+     *
+     * @param type the type/key to associate the {@code instance} with
+     * @param instance the instance
+     */
+    public static void pushDefaultContextObject(Class type, Object instance) {
+        PUSH_DEFAULT_OBJECT.accep(type, instance);
+    }
+
+    private static boolean isRestEasy4() {
+        try {
+            return Class.forName("org.jboss.resteasy.core.ResteasyContext") != null;
+        } catch (ClassNotFoundException ignore) {
+            return false;
+        }
+    }
+
+    /**
+     * Only necessary because keycloak-common is constrained to JDK 1.7.
+     */
+    private interface BiConsumer<T, S> {
+        void accep(T p1, S p2);
```

Comment:
Typo in the method name. Probably doesn't worth to block a PR for longer time, but if it's easily possible to fix, it can be nice :)